src/main/java/com/iota/iri/controllers/ApproveeViewModel.java

```diff
@@ -74,7 +74,7 @@ public static ApproveeViewModel load(Tangle tangle, Indexable hash) throws Excep
         }
 
         approveeViewModel = new ApproveeViewModel((Approvee) tangle.load(Approvee.class, hash), hash);
-        if (cache != null) {
+        if (cache != null && approveeViewModel.getHashes().size() > 0) {
```

Comment:
I usually like using the `isEmpty()`
Usually we also need to check that the collection is not null (but in this case it won't ever be null)